Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -8097,16 +8097,17 @@ RefPtr<CSSValue> consumeContain(CSSParserTokenRange& range)
 RefPtr<CSSValue> consumeContainIntrinsicSize(CSSParserTokenRange& range)
 {
     RefPtr<CSSPrimitiveValue> autoValue;
-    if (range.peek().type() == IdentToken) {
-        switch (range.peek().id()) {
-        case CSSValueNone:
-            return consumeIdent<CSSValueNone>(range);
-        case CSSValueAuto:
-            autoValue = consumeIdent<CSSValueAuto>(range);
-            break;
-        default:
-            return nullptr;
-        }
+    if (range.peek().type() == IdentToken)
+        autoValue = consumeIdent<CSSValueAuto>(range);
+    if (range.atEnd())
+        return nullptr;
+    RefPtr<CSSPrimitiveValue> noneValue;
+    if (range.peek().type() == IdentToken)
+        noneValue = consumeIdent<CSSValueNone>(range);
+    if (noneValue) {
+        if (autoValue)
+            return CSSValueList::createSpaceSeparated(autoValue.releaseNonNull(), noneValue.releaseNonNull());
+        return noneValue;
     }
```

Comment:
I'd rewrite this parsing function to something like this. Also we can use `CSSValuePair` instead of `CSSValueList` here.

```cpp
RefPtr<CSSPrimitiveValue> autoValue;
if (autoValue = consumeIdent<CSSValueAuto>(range)) {
    if (range.atEnd())
        return nullptr;
}

if (auto noneValue = consumeIdent<CSSValueNone>(range)) {
    if (autoValue)
        return CSSValuePair::create(autoValue.releaseNonNull(), noneValue.releaseNonNull());
    return noneValue;
}

if (auto lengthValue = consumeLength(range, HTMLStandardMode, ValueRange::NonNegative)) {
    if (autoValue)
        return CSSValuePair::create(autoValue.releaseNonNull(), lengthValue.releaseNonNull());
    return lengthValue;
}
```

Reply:
Done.